### ISSUE TITLE
Let's go, Pikachu! It's 0.10 and Ruby 3.x time

### ISSUE THREAD

**Issue by apjanke** [apjanke/ronn-ng#103]
This issue is to track my work on getting a v0.10 release out. [Milestone is 0.10.1](https://github.com/apjanke/ronn-ng/milestone/11). (Because there was a bogus 0.10.0 published, so can't use that x.y.0 version.)

My plan is to roll up all the bug fixes and features which have already been merged to master or are "90%" there, bump all the gem deps, and make a new release with that, without waiting for any other pending features or bugfixes.

Important fixes needed:

* [X] https://github.com/apjanke/ronn-ng/issues/102
* [X] https://github.com/apjanke/ronn-ng/issues/107
* [X] https://github.com/apjanke/ronn-ng/issues/100
* [X] https://github.com/apjanke/ronn-ng/issues/87
* [X] https://github.com/apjanke/ronn-ng/issues/65

Let's also do:

* [X] Add prerelease Homebrew formula, for testing that packaging
  * Done in https://github.com/apjanke/homebrew-ronn-ng/commit/f3ce694e9d591ae499d5b285851810baea2f24d7
* [X] Test 0.10.1 prerelease on Intel Mac
  * [X] Install Homebrew on angharad, and switch to new dotfiles
 
Then:

* [X] Publish golden 0.10.1 release
  * Done: [Ronn-NG 0.10.1 on GitHub Releases](https://github.com/apjanke/ronn-ng/releases/tag/v0.10.1)

See the [0.10.1 milestone](https://github.com/apjanke/ronn-ng/milestone/11) for more issues.
 
Hope to satisfy:

* [x] https://github.com/apjanke/ronn-ng/pull/51#issuecomment-1733484890

**Comment by apjanke** [comment]
Work note:

Created a new `WIP/lets-go-pikachu-0.10` branch for this work leading up to an 0.10 release. Working off the `doc-project/Developer-Guide.md` file for doing this release.

The `ronn-ng.gemspec` and other files currently have "0.10.1.pre3" as their version. Bumped that to "0.10.1.pre4". I don't recall there being an 0.10.0 release, but if there's documentation in the repo of it, I'm gonna defer to that. (And the CHANGES file says that was a RubyGems publishing oopse; that sounds pretty plausible.)

To bump the deps, I deleted `Gemfile.lock` locally and then ran `bundle install`.

Got to the "Regenerate the man pages with `bundle exec rake man`" step in the build process, and hit this, on angharad, my Mac desktop:

```
$ bundle exec rake man
env: ruby3.2: No such file or directory
```

I'm not sure what this means. Why does `rake man` know to call `ruby3.2` instead of plain `ruby`? And why don't I have that on my machine? Nothing in the source code seems to be specifying Ruby 3.2 per se.

```
[ronn-ng] $ pwd
/Users/janke/repos/ronn-ng
[ronn-ng] $ grep -r 3\\.2 .
./Gemfile.lock:    parser (3.2.2.4)
./Gemfile.lock:    rexml (3.2.6)
./Gemfile.lock:      parser (>= 3.2.1.0)
[ronn-ng] $
```

**Comment by apjanke** [comment]
Looks like Ruby 3.2 is available in MacPorts as the `ruby32` package. did a `port install ruby32` on angharad.

That took a while. Apparently it has a dependency on rust, and that's a big compile? Waiting for it to finish. I'll post an update tomorrow.

**Comment by apjanke** [comment]
I built and published 0.10.1.pre4 and pre6 releases. Tests are passing on Fedora using the new "test against RPM supplied dependency gems" technique described here (https://github.com/apjanke/ronn-ng/commit/dd2ceebb9e88695a7a8c5042fcc5eb8610da10bf). All the bugs in the 0.10.1 milestone are closed. Going to publish it today.

**Comment by apjanke** [comment]
All done here, I think! I published the 0.10.1 release [on GitHub Releases](https://github.com/apjanke/ronn-ng/releases/tag/v0.10.1) and [on RubyGems](https://rubygems.org/gems/ronn-ng/versions/0.10.1) just now.

**Comment by apjanke** [comment]
For future reference, here's a copy of my local notes for this task.

[ronn-ng 23Q4 fixup notes.md](https://github.com/apjanke/ronn-ng/files/13857113/ronn-ng.23Q4.fixup.notes.md)
